crates/ruff/src/rules/pylint/rules/nested_min_max.rs

```diff
@@ -0,0 +1,104 @@
+use ruff_python_ast::helpers::{has_comments, unparse_expr};
+use ruff_text_size::TextSize;
+use rustpython_parser::ast::{Expr, ExprKind, Keyword};
+
+use ruff_diagnostics::{AlwaysAutofixableViolation, Diagnostic, Edit};
+use ruff_macros::{derive_message_formats, violation};
+
+use crate::{checkers::ast::Checker, registry::AsRule};
+
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub enum NestedMinMaxFunc {
+    Min,
+    Max,
+}
+
+impl NestedMinMaxFunc {
+    /// Returns a value if this is a min() or max() call.
+    fn from_func(func: &Expr) -> Option<NestedMinMaxFunc> {
+        match func.node() {
+            ExprKind::Name { id, .. } if id == "min" => Some(NestedMinMaxFunc::Min),
+            ExprKind::Name { id, .. } if id == "max" => Some(NestedMinMaxFunc::Max),
+            _ => None,
+        }
+    }
+
+    /// Returns true if the passed expr is a call to the same function as self.
+    fn is_call(self, expr: &Expr) -> bool {
+        matches!(expr.node(), ExprKind::Call { func, ..} if NestedMinMaxFunc::from_func(func) == Some(self))
+    }
+}
+
+impl std::fmt::Display for NestedMinMaxFunc {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            NestedMinMaxFunc::Min => write!(f, "min()"),
+            NestedMinMaxFunc::Max => write!(f, "max()"),
+        }
+    }
+}
+
+#[violation]
+pub struct NestedMinMax(NestedMinMaxFunc);
+
+// XXX this need to be better.
+impl AlwaysAutofixableViolation for NestedMinMax {
```

Comment:
This rule should no longer implement `AlwaysAutofixableViolation` because it doesn't provide a fix if there are comments.

Reply:
Fixed